File: backend/controllers/inventory_controller.py
from flask import Blueprint, request, jsonify
from models import InventoryModel

inventory_bp = Blueprint('inventory', __name__)

@inventory_bp.route('/api/inventory', methods=['GET'])
def get_inventory():
    inventory = InventoryModel.get_inventory()
    return jsonify(inventory)

@inventory_bp.route('/api/inventory/add', methods=['POST'])
def add_item():
    data = request.json
    item = InventoryModel.add_item(data['name'], data['price'], data['quantity'], data['category'])
    return jsonify({'success': True, 'item': item}), 201

@inventory_bp.route('/api/inventory/update', methods=['POST'])
def update_item():
    data = request.json
    item = InventoryModel.update_item(data['id'], data['name'], data['price'], data['quantity'], data['category'])
    if item:
        return jsonify({'success': True, 'item': item})
    return jsonify({'success': False, 'message': 'Item not found'}), 404

@inventory_bp.route('/api/inventory/delete', methods=['POST'])
def delete_item():
    data = request.json
    success = InventoryModel.delete_item(data['id'])
    if success:
        return jsonify({'success': True, 'message': 'Item deleted'})
    return jsonify({'success': False, 'message': 'Item not found'}), 404

File: backend/src/controllers/inventoryController.js
const inventoryService = require("../services/inventoryService");

async function getInventory(req, res) {
  const inventory = await inventoryService.listInventory();
  res.json(inventory);
}

async function addInventoryItem(req, res) {
  const item = await inventoryService.createInventoryItem(req.body);
  res.status(201).json({ success: true, item });
}

async function updateInventoryItem(req, res) {
  const item = await inventoryService.updateInventoryItem(req.body);
  if (!item) {
    res.status(404).json({ success: false, message: "Item not found" });
    return;
  }

  res.json({ success: true, item });
}

async function deleteInventoryItem(req, res) {
  const deleted = await inventoryService.deleteInventoryItem(req.body.id);
  if (!deleted) {
    res.status(404).json({ success: false, message: "Item not found" });
    return;
  }

  res.json({ success: true, message: "Item deleted" });
}

module.exports = {
  addInventoryItem,
  deleteInventoryItem,
  getInventory,
  updateInventoryItem
};

